Add window/fulscreen toggle

spaghetticodefactory opened the pull request:
PR for #6 

- Added window mode/fullscreen mode toggle functions
- Connected all signals in code.

quellus commented:
How feasible is to add a toggle to the menu in the app as well? Should this be a separate issue?

spaghetticodefactory commented:
> How feasible is to add a toggle to the menu in the app as well? Should this be a separate issue?

- refactored everything to a `WindowManager` singleton.
- added toggle button to app menu
- cleaned up all signals from GUI to code
- cleaned up function names
- cleaned up label names
- added amplifier & input compression for a more stable input signal. 
- added necessary gui for gain control
- added (debug) label for current input gain display to app menu

Sorry for the huge PR. I should've branched it properly. 

This PR also implements #2

quellus commented:
Committed part of this PR in #45
